zml/module.zig

```diff
@@ -731,8 +757,12 @@ const BaseExe = struct {
     args_buffer_count: u32,
     /// Number of buffers in result.
     result_buffer_count: u32,
+    /// Shapes of buffers in result.
+    result_buffer_shapes: []Shape,
     /// Num devices used (>1 for sharded executable)
     num_devices: u8,
+    /// Allocator backing result_buffer_shapes and deinit by ExeWithWeights
+    _allocator: std.heap.ArenaAllocator,
```

Comment:
For future PR:

what I wanted was to only need to allocate once.
But since `BaseExe` needs to allocate, we know have two allocations.
So we should instead allocate all the slices of `ExeWithWeight` inside `BaseExe`.

Then `prepare` doesn't need to allocate, it just fills out the slices with the model buffers.
And `ExeWithWeight` is just the typed version of `BaseExe`.

Reply:
Tracked down in our issues tracker